mlx/ops.cpp

```diff
@@ -718,6 +718,39 @@ array stack(const std::vector<array>& arrays, StreamOrDevice s /* = {} */) {
   return stack(arrays, 0, s);
 }
 
+/** array repeat with axis */
+array repeat(const array& arr, int repeats, int axis, StreamOrDevice s) {
+  axis = normalize_axis(axis, arr.ndim());
+
+  if (repeats <= 0) {
+    std::vector<int> new_shape(arr.shape());
+    new_shape[axis] = repeats > 0 ? repeats : 0;
+    return zeros(new_shape, arr.dtype(), s);
+  }
```

Comment:
- Throw on negative repeats (as in NumPy) and add a test case to check that it throws.
- For `repeats==0` return an empty array with just one dimensions. Which I think you can do with `array({}, arr.dtype())`